include/load_data.hpp

```diff
@@ -8,9 +8,17 @@
 class Dataloader
 {
 public:
+  // name of the dataset we want to use
   std::string dataset_name;
```

Comment:
If we have getters and setters for these fields, then these should be private and not public.